core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -350,20 +350,16 @@ private[spark] class TaskSchedulerImpl(
                 removeExecutor(execId, reason.get)
                 failedExecutor = Some(execId)
               }
+            }
+            if (TaskState.isFinished(state)) {
+              cleanupTaskState(tid)
```

Comment:
ok last comment: do you think it's more readable to structure this code as:

`if (TaskState.isFinished(state) {
  if (state == TaskState.LOST) {
    <existing code>
    taskResultGetter.enqueueFailed
  } else if (SET(TaskState.FAILED, TaskState.KILLED).contains(state)) {
               taskResultGetter.enqueueFailedTask(taskSet, tid, state, serializedData)
  } else if (state == TaskState.FINISHED) {
       taskResultGetter.enqueueSuccessful(...)
  }

  cleanupTaskState
  taskSet.removeRunningTask
}`